subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -341,6 +341,16 @@ static bool schedule_send(struct bt_mesh_ext_adv *adv)
 	     adv->tags & BT_MESH_ADV_TAG_BIT_FRIEND) ||
 	    (CONFIG_BT_MESH_RELAY_ADV_SETS > 0 && adv->tags & BT_MESH_ADV_TAG_BIT_RELAY)) {
 		k_work_reschedule(&adv->work, K_NO_WAIT);
+	} else if ((IS_ENABLED(CONFIG_BT_MESH_PB_ADV)) && adv->tags & BT_MESH_ADV_TAG_BIT_PROV) {
+		/* The controller will send the next advertisement immediately.
+		 * Introduce a delay here to avoid sending the next mesh packet closer
+		 * to the previous packet than what's permitted by the specification.
+		 */
+		uint8_t random_delay;
+
+		(void)bt_rand(&random_delay, sizeof(random_delay));
+		delta = k_uptime_delta(&timestamp);
+		k_work_reschedule(&adv->work, K_MSEC(20 + random_delay % 31 - delta));
```

Comment:
can use timestamp as random value, since sending timestamps indeed random.

Just:
k_work_reschedule(work, (20 + (timestamp % 31)) - delta);

Reply:
```
Note
While this function returns time in milliseconds, it does not mean it has millisecond resolution. The actual resolution depends on CONFIG_SYS_CLOCK_TICKS_PER_SEC config option.
```
Doesn't seem this is random.

Reply:
Can use Z_TIMEOUT_MS_TICKS?
This look like random?

Reply:
I'm quite sure it is better to use suitable functionality to get random value. Mesh shouldn't depend on platform hidden implementations. I guess `Z_TIMEOUT_MS_TICKS` depends on platform. Also, if you have for some reason the application time synchronized behavior, it stops being random again. 
Talking about efficiency, it is not important in the place of the delay calculation. The behavior is going to be delayed in any case by design. Couple of tens instructions more or less will not impact at all on the final time (time is in tens of milliseconds).